Implement Base Testing Framework for Python Application

sajosam opened the pull request:
This pull request establishes a foundational testing framework for the Python application to enhance testing capabilities for future development.

**Changes Made:**
Created a basic structure to support:
1. Integration Testing
2. Functional Testing
3. Unit Testing

agberoz commented on app/tests/conftest.py:
```diff
@@ -0,0 +1,98 @@
+from typing import Generator, Any
```
can you move `tests` folder to base directory

agberoz commented on app/tests/conftest.py:
```diff
@@ -0,0 +1,98 @@
+from typing import Generator, Any
+import pytest
```
Add documentation for testing instructions in Readme.md

sajosam commented:
@agberoz 
- Added test file to root directory
- Added readme file for testing
![image](https://github.com/user-attachments/assets/0ffc11bd-62ba-4d64-ab5c-5c1383e99b9c)
![image](https://github.com/user-attachments/assets/f0d49617-cf64-4792-ad98-253e39392da6)

agberoz commented on tests/README.md:
```diff
@@ -0,0 +1,30 @@
+- `integration/`: Contains integration tests (`test_*.py`).
+- `unittest/`: Contains unit tests (`test_*.py`).
+
+## Requirements
```
add dependency with version to requirement.txt

agberoz reviewed (approved):
LGTM
